elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,596 @@
+                        String.format("Unsupported definition type {%s}.", definition.getClass()));
+            }
+        });
+
+        return new GraphQLProjectionInfo(rootProjections, relationshipMap);
+    }
+
+    /**
+     * Root projection would be an operation applied on an single entity class.
+     * The EntityProjection tree would be constructed recursively to add all child projections.
+     *
+     * Currently we only support one root class in each GraphQL {@link OperationDefinition} since
+     * each {@link EntityProjection} only has one entity class.
+     *
+     * For example
+     * <pre>
+     *     Query {
+     *         Book {
+     *             field1
+     *         },
+     *         Book {
+     *             field2
+     *         }
+     *     }
+     * </pre>
+     * Would be a valid GraphQL request, and would be resolved into
+     * <pre>
+     *     Query {
+     *         Book {
+     *             field1,
+     *             field2
+     *         }
+     *     }
+     * </pre>
+     *
+     * <pre>
+     *     Query {
+     *         Book {
+     *             field1
+     *         },
+     *         Author {
+     *             field2
+     *         }
+     *     }
+     * </pre>
+     * Would be an invalid request as two classes-Book and Author-are both queried at root level.
+     *
+     * @param selectionSet a root-level selection set
+     */
+    private void addRootProjection(SelectionSet selectionSet) {
+        List<Selection> selections = selectionSet.getSelections();
+
+        // get the first selection as root selection
+        Selection rootSelection = selections.get(0);
+        if (!(rootSelection instanceof Field)) {
+            throw new InvalidEntityBodyException("Entity selection must be a graphQL field.");
+        }
+
+        String entityName = ((Field) rootSelection).getName();
+        if (SCHEMA.equals(entityName) || TYPE.equals(entityName)) {
+            // '__schema' and '__type' would not be handled by entity projection
+            return;
+        }
+
+        Class<?> entityType = entityDictionary.getEntityClass(entityName);
+        if (entityType == null) {
+            throw new InvalidEntityBodyException(String.format("Unknown entity {%s}.", entityName));
+        }
+
+        Field fields = (Field) rootSelection;
+
+        // merging partial graphql selections, skip(1) because the first selection is already processed
+        selections.stream().skip(1).forEach(selection -> {
+            if (!(selection instanceof Field)) {
+                throw new InvalidEntityBodyException("Entity selection must be a graphQL field.");
+            }
+            if (!entityName.equals(((Field) selection).getName())) {
+                throw new InvalidEntityBodyException("Can't select multiple entities in graphQL QUERY operation.");
+            }
+            fields.getSelectionSet().getSelections().addAll(((Field) selection).getSelectionSet().getSelections());
```

Comment:
```java
fields.getSelectionSet().getSelections().addAll(((Field) selection).getSelectionSet().getSelections());
```
Currently this is not checking field name or alias name, if a Elide field is queried in both graphQL `Field`, it would be queried twice.

Reply:
If two fields, one from each query, have the same name and alias, they would be merged into a single field. If two fields, one from each query, have different name but same alias, they would be ambiguous.